### PR TITLE
Add documental land-user

### PR DESCRIPTION
**Pull request by smartsena**

### CODE REVIEW

File: documental/serializers.py [hunk 1]
```diff
@@ -70,3 +70,39 @@ def to_representation(self, instance):
         return url_document
 
 
+class DocumentosTISerializers(serializers.ModelSerializer):
+    """Serializer to return action category `models.DocumentalDocs` data.
+
+    Data only for the action category linked to DOCUMENTAL_TI.
+    """
+    usuario_id = UsuarioSerializers()
```
Comment (dagnaldo) [inline documental/serializers.py:78]: Missing spaces.
You are doing the same thing every pull request.

**Review by dagnaldo — CHANGES_REQUESTED** [review]

File: documental/serializers.py (hunk 1 above)
Comment (smartsena) [reply documental/serializers.py:78]: ok.

File: documental/views.py [hunk 2]
```diff
@@ -33,10 +33,31 @@ class DocumentalListViews(generics.ListAPIView):
...
+            `serializers.DocumentosTISerializers`.
+        """
+
+        actions_id_land_use = ['101', '102', '103']
+        error_mensag = 'Action not defined in your request.'
+        requested_action = str(self.request.GET.get('acao_id', error_mensag))
+
+        if requested_action in actions_id_land_use:
+            print(f"MapasUsoOcupacaoSolo: {requested_action}")
+            return serializers.MapasUsoOcupacaoSoloSerializers
```
[the remark this answers is not in the document]
Comment (smartsena) [reply documental/views.py:60]: No, one is intended for the "painel\manager_mapasusoocupacaosolo" data and the other for the documental data "painel\manager_documentosti".

File: documental/views.py [hunk 3]
```diff
@@ -33,10 +33,31 @@ class DocumentalListViews(generics.ListAPIView):
...
+        else:
+            print(f"DocumentosTI: {requested_action}")
```
[the remark this answers is not in the document]
Comment (smartsena) [reply documental/views.py:62]: Removed.

File: documental/models.py [hunk 4]
```diff
@@ -18,11 +20,27 @@ class Action(models.Model):
         blank=True,
     )
 
+    # categoria_acao = models.CharField(
```
Comment (dagnaldo) [inline documental/models.py:23]: comments?

File: documental/models.py [hunk 5]
```diff
@@ -18,11 +20,27 @@ class Action(models.Model):
...
+    # descricao = models.CharField(
+    #     _('Description of documental type'),
+    #     max_length=512,
+    #     null=True,
+    #     blank=True,
+    # )
+
     class Meta:
         """"Meta class for `documental.Action` model."""
+        # """"Meta class for `documental.DocsAction` model."""
```
Comment (dagnaldo) [inline documental/models.py:39]: Comments?

File: documental/models.py [hunk 6]
```diff
@@ -18,11 +20,27 @@ class Action(models.Model):
...
         app_label = 'documental'
-        verbose_name = 'Action'
-        verbose_name_plural = 'Actions'
+        verbose_name = 'Action' #'Document Action'
```
Comment (dagnaldo) [inline documental/models.py:41]: Comments?

File: documental/models.py [hunk 7]
```diff
@@ -4,7 +4,9 @@
 
 
 class Action(models.Model):
+# class DocsAction(models.Model):
```
Comment (dagnaldo) [inline documental/models.py:7]: comments?

File: documental/models.py [hunk 8]
```diff
@@ -4,7 +4,9 @@
...
     """Action model data for documental model."""
+    # """DocsAction model data for documental model."""
```
Comment (dagnaldo) [inline documental/models.py:9]: comments?

File: documental/filters.py [hunk 9]
```diff
@@ -22,16 +22,17 @@ class DocumentalDocsFilter(rest_framework.FilterSet):
...
+        * acao_id (int): action identifier to be filtered.
         * co_cr (list): filtering Regional Coordination using code.
         * co_funai (list): filtering Indigenou Lands using Funai code.
         * start_date (str): filtering start date.
         * end_date (str): filtering end date.
+        * map_year (list): filteringend years of the maps.
     """
     
-    acao_id = rest_framework.CharFilter(
+    acao_id = CharInFilter(
```
Comment (dagnaldo) [inline documental/filters.py:33]: PEP8 import errors, it might come from `app.MethodName`

**Review by dagnaldo — CHANGES_REQUESTED** [review]

File: documental/models.py (hunk 4 above)
Comment (smartsena) [reply documental/models.py:23]: ok

File: documental/models.py (hunk 6 above)
Comment (smartsena) [reply documental/models.py:41]: ok

File: documental/models.py (hunk 7 above)
Comment (smartsena) [reply documental/models.py:7]: ok

File: documental/models.py (hunk 8 above)
Comment (smartsena) [reply documental/models.py:9]: ok

File: documental/filters.py (hunk 9 above)
Comment (smartsena) [reply documental/filters.py:33]: Not modified. The return of the Foreign key model is charfield.

File: documental/views.py [hunk 10]
```diff
@@ -19,24 +19,46 @@ class AuthModelMix:
...
+            `serializers.DocumentosTISerializers`.
+        """
+
+        actions_id_land_use = [11, 12, 13,]
+        error_mensag = 'Action not defined in your request.'
+        requested_action = self.request.GET.get('id_acao', error_mensag)
+        requested_action = list(map(int,requested_action.split(',')))
+        for id_action, action in enumerate(requested_action):
+            if action in actions_id_land_use:
+                print("MapasUsoOcupacaoSoloSerializers")
```
Comment (dagnaldo) [inline documental/views.py:60]: Remove these prints.

**Review by dagnaldo — CHANGES_REQUESTED** [review]

File: documental/views.py (hunk 10 above)
Comment (smartsena) [reply documental/views.py:60]: Removed.

**Review by dagnaldo — CHANGES_REQUESTED** [review]
take a simple look, please.

File: documental/views.py [hunk 11]
```diff
@@ -18,25 +18,47 @@ class AuthModelMix:
...
+
+        Returns one serializers class to `views.ActionListView`
+        
+        Returns:
+            `serializers.MapasUsoOcupacaoSoloSerializers` or
+            `serializers.DocumentosTISerializers`.
+        """
+
+        actions_id_land_use = [11, 12, 13,]
+        error_mensag = 'Action not defined in your request.'
```
[the remark this answers is not in the document]
Comment (dagnaldo) [reply documental/views.py:55]: It is not removed, yet... take a look.

**Review by dagnaldo — CHANGES_REQUESTED** [review]

File: documental/views.py (hunk 11 above)
[the remark this answers is not in the document]
Comment (smartsena) [reply documental/views.py]: Sorry, removed now.

File: documental/views.py [hunk 12]
```diff
@@ -18,25 +18,46 @@ class AuthModelMix:
...
+        Returns:
+            `serializers.MapasUsoOcupacaoSoloSerializers` or
+            `serializers.DocumentosTISerializers`.
+        """
+
+        actions_id_land_use = [11, 12, 13,]
+        requested_action = self.request.GET.get('id_acao')
+        requested_action = list(map(int,requested_action.split(',')))
+
+        for id_action, action in enumerate(requested_action):
```
Comment (dagnaldo) [inline documental/views.py:58]: Logic error, please, check it again.

**Review by dagnaldo — CHANGES_REQUESTED** [review]

**Review by dagnaldo — APPROVED** [review]

**Pull request merged**